GetStream/stream-chat-swift
# [Background Mapping] Ensure new pages are not requested until the current page has been returned

### 🔗 Issue Links

https://github.com/GetStream/ios-issues-tracking/issues/637

### 🎯 Goal

We should not be requesting more than a page at a time when paginating in a Channel

### 📝 Summary

There was no way for us to guarantee that we would not request a new page when already getting one.

### 🛠 Implementation

StatefulScrollViewPaginationHandler  now keeps track of whether there is any active page being fetched, and would not allow a new one until the completion of `onNewTopPage`/`onNewBottomPage` is called.

### 🧪 Manual Testing Notes
1. Open a channel with many pages
2. Scroll fast to the top

Expected result:
The same page should not be requested twice

### ☑️ Contributor Checklist


### 🎁 Meme

![](https://media.giphy.com/media/4LG8mYIacy4Pn610rM/giphy.gif)
